enhancements/cluster-logging/logs-based-alerts.md

```diff
@@ -0,0 +1,300 @@
+---
+title: logs-based-alerts
+authors:
+  - "@periklis"
+reviewers:
+  - "@alanconway"
+  - "@xperimental"
+  - "@jcantrill"
+  - "@simonpasquier"
+  - "@fpetkovski"
+approvers:
+  - "@alanconway"
+  - "@simonpasquier"
+  - "@fpetkovski"
+api-approvers:
+  - "@alanconway"
+creation-date: "2022-05-16"
+last-updated: "2022-05-19"
+tracking-link:
+  - https://issues.redhat.com/browse/LOG-2510
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Logs-based Alerts
+
+## Summary
+
+This document proposes a solution and a delivery plan for pushing logs-based alerts from OpenShift Logging Loki to Cluster-Monitoring AlertManager. It complements the existing [user-workload monitoring stack][uwm-docs] and [alerting stack][alerting-docs], enabling a full self-service experience for platform and workload logs-based monitoring.
```

Comment:
There's no kind of flow control in Alertmanager as of today. I've checked the Loki code base quickly and it seems to use the same techniques as Prometheus to avoid hammering Alertmanager:
* It has a default delay of 1 minute before resending the same alert to Alertmanager.
* It keeps alerts in an in-memory queue (up to 10,000 alerts). If Alertmanager can't keep up on the receiving side, older alerts are dropped.

I would expect that Loki is impacted as well when we hit this "too many alerts" issue.

Reply:
Literally Loki replicates 1:1 the Prometheus code if it is not using it directly :P